### PR TITLE
LPS-203146 Enable search ObjectEntries by relationship ObjectField when the title ObjectField of the parent ObjectDefinition is different from it ID

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPS-203146

### CODE REVIEW

**Comment by p-albuquerque** [comment]
ci:test:relevant

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 1]
```diff
@@ -1963,6 +1969,67 @@ public void testGetObjectEntries() throws Exception {
...
+			objectDefinitionLocalService.updateObjectDefinition(
+				_objectDefinition1);
+
+		testGetObjectEntries(
+			HashMapBuilder.put(
+				"search", "Able"
+			).build(),
+			childObjectEntry1);
+
+		objectField = ObjectFieldUtil.addCustomObjectField(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:1989]: Can we move this field creation to **setUp** method?

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 2]
```diff
@@ -1963,6 +1969,67 @@ public void testGetObjectEntries() throws Exception {
 				"search", "aa"
 			).build(),
 			childObjectEntry1, childObjectEntry2);
+
+		ObjectField objectField = objectFieldLocalService.fetchObjectField(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:1973]: What do you think to move this scenario to a private method (**testSearchByRelationshipField** or **testSearchByRelatedObjectEntry**)? or at least put a comment here to give context.

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 3]
```diff
@@ -1963,6 +1969,67 @@ public void testGetObjectEntries() throws Exception {
...
+			).build());
+
+		_objectDefinition1.setTitleObjectFieldId(
+			objectField.getObjectFieldId());
+
+		_objectDefinition1 =
+			objectDefinitionLocalService.updateObjectDefinition(
+				_objectDefinition1);
+
+		String objectFieldName = objectField.getName();
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:2008]: Is it possible to inline this?

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 4]
```diff
@@ -1963,6 +1969,67 @@ public void testGetObjectEntries() throws Exception {
...
+		childObjectEntry1 = _defaultObjectEntryManager.getObjectEntry(
+			dtoConverterContext, _objectDefinition2, childObjectEntry1.getId());
+
+		testGetObjectEntries(
+			HashMapBuilder.put(
+				"search", "Baker"
+			).build(),
+			childObjectEntry1);
+
+		childObjectEntry2 = _defaultObjectEntryManager.getObjectEntry(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:2030]: Is this really necessary?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/entry/util/ObjectEntrySearchUtil.java [hunk 5]
```diff
@@ -54,12 +54,11 @@ else if (dbType.equals(ObjectFieldConstants.DB_TYPE_INTEGER) ||
 	}
 
 	public static Predicate getRelatedModelsPredicate(
-		DynamicObjectDefinitionTable dynamicObjectDefinitionTable,
-		ObjectDefinition objectDefinition,
+		Table<?> table, ObjectDefinition objectDefinition,
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/entry/util/ObjectEntrySearchUtil.java:57]: Can you please sort the parameters?

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by guilhermedcamacho** [comment]
ci:test:object

**Comment by guilhermedcamacho** [comment]
ci:forward:force

**Pull request closed without merging**